src/nnet3/nnet-optimize.cc

```diff
@@ -878,6 +938,12 @@ const NnetComputation* CachingOptimizingCompiler::Compile(
       checker.Check();
     }
     computation_.ComputeCudaIndexes();
+    UpdateCache(true);
+  } else {
+    // if found, get computation and update access queue
+    request_ = *(cit->first);
+    computation_ = *(cit->second.first);
```

Comment:
Yes- delete them there; and whenever you remove them from the map in the
UpdateCache function or whatever you decide to call it.  Get it to a state
where valgrind does not report any 'definitely lost'
Dan


On Sat, Sep 5, 2015 at 1:59 AM, Xingyu Na <notifications@github.com> wrote:

> In src/nnet3/nnet-optimize.cc
> <https://github.com/kaldi-asr/kaldi/pull/120#discussion_r38808116>:
>
> > @@ -878,6 +938,12 @@ const NnetComputation* CachingOptimizingCompiler::Compile(
> >        checker.Check();
> >      }
> >      computation_.ComputeCudaIndexes();
> > +    UpdateCache(true);
> > +  } else {
> > +    // if found, get computation and update access queue
> > +    request_ = *(cit->first);
> > +    computation_ = *(cit->second.first);
>
> BTW, should I delete the newed computations in a deconstructer.
> … <#14f9c16319fa2ead_>
> On Sep 5, 2015, at 12:20, Daniel Povey <notifications@github.com> wrote:
> In src/nnet3/nnet-optimize.cc: > @@ -878,6 +938,12 @@ const
> NnetComputation* CachingOptimizingCompiler::Compile( > checker.Check(); > }
> > computation_.ComputeCudaIndexes(); > + UpdateCache(true); > + } else { >
> + // if found, get computation and update access queue > + request_ =
> *(cit->first); > + computation_ = *(cit->second.first); Yes, remove them. …
> — Reply to this email directly or view it on GitHub.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/120/files#r38808116>.
>